### PR TITLE
Do flush 0-valued counters

### PR DESCRIPTION
**Pull request by haltwise**

### CODE REVIEW

File: stats_test.go [hunk 1]
```diff
@@ -276,33 +276,31 @@ func TestPerInstanceStats(t *testing.T) {
 		},
 	}
 
-	sink := mock.NewSink()
-
-	testPerInstanceMethods := func(t *testing.T, scope Scope) {
+	testPerInstanceMethods := func(t *testing.T, setupScope func(Scope) Scope) {
```
Comment (charlievieth) [inline stats_test.go:279]: Can we add a test that explicitly checks that we flush zero valued counters? Could be super simple, I just wanna prevent regressions.

**Review by charlievieth — DISMISSED** [review]
LGTM

**Review by charlievieth — APPROVED** [review]
LGTM
